lib/bundler/lockfile_parser.rb

```diff
@@ -101,17 +101,16 @@ def warn_for_outdated_bundler_version
       return unless bundler_version
       prerelease_text = bundler_version.prerelease? ? " --pre" : ""
       current_version = Gem::Version.create(Bundler::VERSION)
-      case current_version.segments.first <=> bundler_version.segments.first
-      when -1
-        raise LockfileError, "You must use Bundler #{bundler_version.segments.first} or greater with this lockfile."
-      when 0
-        if current_version < bundler_version
-          Bundler.ui.warn "Warning: the running version of Bundler (#{current_version}) is older " \
-               "than the version that created the lockfile (#{bundler_version}). We suggest you to " \
-               "upgrade to the version that created the lockfile by running `gem install " \
-               "bundler:#{bundler_version}#{prerelease_text}`.\n"
-        end
+      current_major_version = current_version.segments.first
+      major_bundler_version = bundler_version.segments.first
```

Comment:
`bundler_version` being the `BUNDLED WITH` version is a very confusing name (it's not Bundler's version!).
The code was already like that, but it definitely doesn't make it easy to understand what happens.

Reply:
Yes, absolutely agree. I left it there because it was already like that, but should be changed. I can do that.